pkg/allocation/converters/converter_test.go

```diff
@@ -203,20 +203,25 @@ func TestConvertAllocationRequestToGameServerAllocation(t *testing.T) {
 							},
 						},
 					},
-					Required: allocationv1.GameServerSelector{LabelSelector: metav1.LabelSelector{
-						MatchLabels: map[string]string{
-							"c": "d",
+					Required: allocationv1.GameServerSelector{
```

Comment:
For this specific test group `"all fiends are set"` - rather than add `GameServerState` to everything, let's add a 

```
features: fmt.Sprintf("%s=false&%s=false", runtime.FeaturePlayerAllocationFilter, runtime.FeatureStateAllocationFilter),
```

To [this block](https://github.com/googleforgames/agones/pull/2695/files#diff-5b3da3128a01faaf200ec681805dab5edced5aeb32314af0647488b6a95b50c7R146). This test block is a repeat of `"all fields are set (StateAllocationFilter, PlayerAllocationFilter)"` but with the feature flags flipped (which by default they used to be) - so we should maintain the unit tests of the Alpha features.

We should do the same for any other tests that follow the same pattern (specifically turn off the feature flag for the original version of the test).

Reply:
I fixed tests and I also added a missing test.
Please let me know if I'm doing wrong.